[T12-4] PDF++

ptvrajsk opened the pull request:
@jetkan-yk
@WeiTangLau
@lormee94

nhs-work commented:
Feedback for v1.1:

Github project page:
 
![image](https://user-images.githubusercontent.com/35650824/54073187-0c0ee400-42bf-11e9-898d-db6e0e066178.png)

Please update the name as well as the link to your github project.

Readme:
Will this be your final UI mockup? It looks like a very rough sketch at this point in time.
Please acknowledge AB-4.
Good that all the badges have been updated accordingly.

About us:
Please indicate which components each member is in charge of (e.g. In charge of Model, UI, Storage, etc.).

User Guide/Developer Guide:
Comments will be provided in separate PDF documents that will be uploaded to Luminus.

nhs-work commented:
Feedback for v1.2:

Overall good job on v1.2!

The implementation section is filled with user specific details that are lacking from the current user guide, do consider including them into the user guide instead. It also does not delve deep enough into the component/class interactions occurring for each feature, why you chose such an implementation and alternatives.

![image](https://user-images.githubusercontent.com/35650824/54881096-44015400-4e87-11e9-8bf2-4fd2cc5ef5de.png)
 

This was a good attempt at an activity diagram and the basic idea is present. All that is needed to follow the notations taught during the lecture/in the course website.

![image](https://user-images.githubusercontent.com/35650824/54881097-4663ae00-4e87-11e9-840c-4350eea3cfe9.png)

 
Do note that certain images appear to be missing/misnamed.

I have briefly gone through the code committed thus far and noted that there are large chunks of previous code that has been commented out but not removed. Please do remove them if they are not required, you can always retrieve them from a previous commit if necessary (This is why we have revision control!).

ptvrajsk committed:
Merge branch 'master' into AutoTest

Reference #217

ptvrajsk committed:
Checkstyle Fixes + Code Optimisation

Moving some variables to CommandTestUtil. Reference #217

ptvrajsk committed:
JsonPdfBookStorageTest Working

Revamped and got the above test to work. Reference #217.

WeiTangLau committed:
Tracking for commands in build.gradle

WeiTangLau committed:
travis check

WeiTangLau committed:
travis

lormee94 committed:
(v1.4) DG updates

WeiTangLau committed:
travis

WeiTangLau committed:
travis test

WeiTangLau committed:
(v1.4) DeleteCommandParserTest

Test case for DeleteCommandParser.
see: #217

WeiTangLau committed:
(v1.4) EditCommandParserTest

Provided test cases for EditCommandParser.
see : #217

ptvrajsk committed:
(v1.4) Fixed JsonSerializablePdfBookTest

With some changes to the test json files, managed to fix the
serializable pdf book test. Reference: #217

ptvrajsk committed:
(v1.4) Fixed JsonUserPrefsStorageTest

Minor changes to file, currently in passing state. Reference: #217

ptvrajsk committed:
(v1.4) StorageManagerTest working

StorageManagerTest working out of the box, implemented some naming
convention changes across the storage unit and included the storage
tests in build.gradle. Expect Coverage increase. Reference: #217

ptvrajsk committed:
Travis Fixes

WeiTangLau committed:
(v1.4) Parser Command

Test cases for the parser command.

ptvrajsk committed:
(v1.4) Removed SerializablePdfBookTest

Json files require legitimate files hence, unable to preset them in the
test json file that is uploaded to travis. Travis would always show it
as failing. Hence removed that test. Reference: #217

ptvrajsk committed:
(v1.4) TagCommand

TagCommand brought to skeletal working state. Reference: #217

ptvrajsk committed:
Checkstyle Fix

ptvrajsk committed:
Checkstyle Fixes

jetkan-yk committed:
Merge pull request #279 from ptvrajsk/AutoTest

Auto Test Updates [READY TO MERGE]

lormee94 committed:
(v1.4) DG rework
- Add Command and related diagrams

WeiTangLau committed:
(v1.4) PdfBookParserTest

Provided test cases for PdfBookParser.
See: #217

lormee94 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main into DG

ptvrajsk committed:
Merge branch 'master' into master

lormee94 committed:
Travis

ptvrajsk committed:
Merge pull request #277 from WeiTangLau/master

Travis for commands

lormee94 committed:
Travis

ptvrajsk committed:
Merge branch 'master' into v1.4IntegrationTests

WeiTangLau committed:
Merge branch 'master' into v1.4IntegrationTests

lormee94 committed:
Travis

WeiTangLau committed:
(v1.4) travis

ptvrajsk committed:
Merge pull request #280 from lormee94/DG

(v1.4 New format of DG - Implementation)

WeiTangLau committed:
Travis test

WeiTangLau committed:
Merge branch 'master' into v1.4IntegrationTests

WeiTangLau committed:
travis

WeiTangLau committed:
travis

ptvrajsk committed:
(v1.4) TagCommandParserTest Complete

Implemented a working version of the TagCommandParserTest.
Reference #217.

WeiTangLau committed:
travis

ptvrajsk committed:
Merge branch 'master' into AutoTest

WeiTangLau committed:
travis

lormee94 committed:
(v1.4) DG Activity Diagram
- Clear
- Decrypt

lormee94 committed:
Merge pull request #278 from WeiTangLau/v1.4IntegrationTests

(v1.4) Parser Tests

lormee94 committed:
Merge pull request #281 from ptvrajsk/AutoTest

Auto Test Update

WeiTangLau committed:
(v1.4) ParserUtil Test cases

ptvrajsk committed:
Merge pull request #282 from WeiTangLau/master

(v1.4) ParserUtil Test cases

lormee94 committed:
(v1.4) DG update diagram
- Encrypt

WeiTangLau committed:
(v1.4) Logic Tests

WeiTangLau committed:
travis

WeiTangLau committed:
satify Travis

NOTE The test in DeleteCommandTest line 67 has been commented out for
travis test. Have to come back and find out the bug.

WeiTangLau committed:
travis

WeiTangLau committed:
travis

lormee94 committed:
Merge pull request #283 from WeiTangLau/master

(v1.4) Logic Tests

lormee94 committed:
(v1.4) DG update Diagram
- Find
- Merge

lormee94 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main into DG

ptvrajsk committed:
DG Update

ptvrajsk committed:
(v1.4) DG Updates

Find Feature DG Documentation.

ptvrajsk committed:
Merge branch 'master' into AutoTest

jetkan-yk committed:
Update DeadlineCommand Autotest

jetkan-yk committed:
Update DeadlineCommand Autotest

jetkan-yk committed:
Update DeadlineCommand Test

Track issue: CS2103-AY1819S2-T12-4#217

jetkan-yk committed:
Checkstyle fixes

jetkan-yk committed:
Update DeadlineCommandParser Autotests

Track issue: #217

WeiTangLau committed:
Merge pull request #286 from jetkan-yk/test/DeadlineCommandParser

Update DeadlineCommandParser Autotest

jetkan-yk committed:
(v1.4) Bug fix on Tag Command

Fixed no error message thrown when removing a tag that does not exist.

Resolves: #226
Resolves: #236

jetkan-yk committed:
Update TagCommand Autotest

jetkan-yk committed:
Update TagCommand Autotest

WeiTangLau committed:
(v1.4) Refactoring to Pdf

Refactoring from AddressBook to PdfBook.

WeiTangLau committed:
Merge pull request #288 from jetkan-yk/bug/Tag

(v1.4) Bug fix on Tag Command

WeiTangLau committed:
Travis check

WeiTangLau committed:
Commit to merge PR from upstream

WeiTangLau committed:
Merge branch 'master' into v1.4Refactor

jetkan-yk committed:
Update deadlineCommand Autotests

jetkan-yk committed:
Internal branch merge

WeiTangLau committed:
refactor all instance of 'address'

jetkan-yk committed:
Update DeadlineCommand

Remove redundant function in DeadlineAction

WeiTangLau committed:
Travis test

jetkan-yk committed:
Update DeadlineCommandParser Autotest

WeiTangLau committed:
Merge pull request #290 from jetkan-yk/master

Update DeadlineCommand Autotest

WeiTangLau committed:
Conflict resolution for DeadlineCommandTest

jetkan-yk committed:
Merge pull request #289 from WeiTangLau/v1.4Refactor

(v1.4) Refactoring to Pdf

jetkan-yk committed:
Update DeadlineCommand Autotest

jetkan-yk committed:
Merge branch 'master' of github.com:CS2103-AY1819S2-T12-4/main

WeiTangLau committed:
Merge pull request #291 from jetkan-yk/master

Update DeadlineCommand Autotest

WeiTangLau committed:
increase coverage of test

WeiTangLau committed:
(v1.4)TagContainsKeywordsPredicateTest

Provided test cases for TagContainsKeywordsPredicate.
See: #217

WeiTangLau committed:
Travis Test

jetkan-yk committed:
Merge pull request #292 from WeiTangLau/master

increase coverage of test

jetkan-yk committed:
Merge pull request #293 from WeiTangLau/v1.4TagContainsKeywordsPredicateTest

(v1.4) TagContainsKeywordsPredicateTest

WeiTangLau committed:
(v1.4) Refactor EditCommand to RenameCommand

WeiTangLau committed:
Travis check for PdfBookParser

WeiTangLau committed:
Travis test

jetkan-yk committed:
Merge pull request #294 from WeiTangLau/v1.4TagContainsKeywordsPredicateTest

(v1.4) Refactor EditCommand to RenameCommand

jetkan-yk committed:
(v1.4) Debug Tag Command

Added stricter input validation for tag commands.

Resolves: #228

jetkan-yk committed:
Merge branch 'master' of github.com:CS2103-AY1819S2-T12-4/main

jetkan-yk committed:
Checkstyle fixes

jetkan-yk committed:
Debug sort by name command

As mentioned in #287, sort name command now sorts based on
dictionary order instead of ASCII order.

Resolves: #287

ptvrajsk committed:
Travis Fixes

ptvrajsk committed:
Merge branch 'master' into AutoTest

ptvrajsk committed:
Merge pull request #295 from jetkan-yk/master

Debug Tag Command

ptvrajsk committed:
Merge pull request #296 from jetkan-yk/debug/Sort

Debug sort by name command

jetkan-yk committed:
Merge pull request #284 from ptvrajsk/AutoTest

DG Update

WeiTangLau committed:
Modified DG

lormee94 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main into DG

WeiTangLau committed:
(v1.4) Design Diagrams for Developer Guide

Provided the diagrams for part 2: design of Developer Guide.
Note that I have skipped one of the diagrams. Will revisit it.
See: #215

WeiTangLau committed:
case-sensitive RenameCommand

Modified the rename command to support renaming with case-sensitive.
a.pdf can now be renamed to A.pdf.

resolved: #296

lormee94 committed:
(v1.4) DG implementation
- Merge command

WeiTangLau committed:
Resolve RenameCommandTest

WeiTangLau committed:
Minor file modification

WeiTangLau committed:
travis test

WeiTangLau committed:
Merge branch 'master' into v1.4DG

WeiTangLau committed:
travis check

WeiTangLau committed:
Support for both linux and Windows RenameCommand

WeiTangLau committed:
Travis

WeiTangLau committed:
Travis test

WeiTangLau committed:
travis

WeiTangLau committed:
RenameCommand now checks for system.

Additional checks to support for cross platform renaming of files.
See: #297

WeiTangLau committed:
Travis test

WeiTangLau committed:
testing for linux OS

WeiTangLau committed:
travis check

WeiTangLau committed:
appveyor check

ptvrajsk committed:
(v1.4) Remvoved a redundant check in Pdf.java

isFileEncrypted method performs a redundant check in checking if the
file is encrypted. This uses an external API that seems to be unreliable
in some cases. Removed it and streamlined the process to remove that
bug.

lormee94 committed:
(v1.4) DG update

lormee94 committed:
Travis

jetkan-yk committed:
Merge pull request #299 from lormee94/DG

(v1.4) Updates to DG + Others

jetkan-yk committed:
Merge pull request #300 from ptvrajsk/DG

(v1.4) Remvoved a redundant check in Pdf.java [Can Merge, Decrease coverage by 0.01%]

jetkan-yk committed:
Merge pull request #298 from WeiTangLau/v1.4DG

Developer Guide Modification

jetkan-yk committed:
Update UG and DG

jetkan-yk committed:
Create DeadlineCommandActivityDiagram

jetkan-yk committed:
Checkstyle fixes

WeiTangLau committed:
(v1.4) DG for File Protection section

Provided the Encrypt and Decrypt command sequence diagram.
Made minor adjustments.

See: #215

WeiTangLau committed:
(v1.4) Developer Guide for File Protection

Completed the developer guide for File Protection section.

WeiTangLau committed:
Travis test

WeiTangLau committed:
Merge pull request #301 from jetkan-yk/master

Add select command in UG

WeiTangLau committed:
Merge branch 'master' into v1.4FileProtectionSequenceDiagram

WeiTangLau committed:
UG Modification for File Protection

ptvrajsk committed:
Merge pull request #302 from WeiTangLau/v1.4FileProtectionSequenceDiagram

v1.4 Developer Guide File Protection section

WeiTangLau committed:
(v1.4) UG modification for File Protection section

WeiTangLau committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main

ptvrajsk committed:
(v1.4) Improvements in Find Feature DG

Improved the diagrams required for the Find feature and the explanations
as well.

ptvrajsk committed:
Merge branch 'master' into DG

ptvrajsk committed:
Checkstyle Fixes

WeiTangLau committed:
Merge pull request #303 from ptvrajsk/DG

DG Update

WeiTangLau committed:
(v1.4) UG Guide for File Protection

Provided description for Encrypt Command.

WeiTangLau committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main

WeiTangLau committed:
File Protection UG images

WeiTangLau committed:
Merge branch 'master' into v1.4FileProtectionUserGuide

ptvrajsk committed:
(v1.4) UG Update

Input a "find" section to illustrate the usage of the find command.

WeiTangLau committed:
Merge pull request #304 from ptvrajsk/DG

(v1.4) UG Update

WeiTangLau committed:
(v1.4) UG for FileProtection

Documentation on UG for FileProtection has been completed.
See: #214

WeiTangLau committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main

WeiTangLau committed:
Merge branch 'master' into v1.4FileProtectionUserGuide

ptvrajsk committed:
Merge pull request #305 from WeiTangLau/v1.4FileProtectionUserGuide

v1.4 FileProtection User Guide

WeiTangLau committed:
(v1.4) Architecture diagram adjustments

ptvrajsk committed:
Merge pull request #306 from WeiTangLau/v1.4FileProtectionUserGuide

(v1.4) Architecture diagram adjustments

jetkan-yk committed:
Refactor to PDF++

ptvrajsk committed:
DG Update: Select

ptvrajsk committed:
DG Update: Sort

jetkan-yk committed:
Refactor to PDF++

ptvrajsk committed:
DG: Update Tag

jetkan-yk committed:
Bug fixes for refactoring to PDF++

jetkan-yk committed:
Checkstyle fixes

ptvrajsk committed:
DG: Update Filter

jetkan-yk committed:
Minor refactor

jetkan-yk committed:
Merge pull request #308 from ptvrajsk/DG

DG Update [CAN MERGE NOW]

WeiTangLau committed:
(v1.4) Pdf Class Diagram

I have included the Pdf Class diagram in the DG under the model section. This
will provide the developers with a better understanding on the most
important class, Pdf.

ptvrajsk committed:
UG & PPP Update

ptvrajsk committed:
Merge pull request #307 from jetkan-yk/master

Refactor to PDF++

WeiTangLau committed:
Merge pull request #309 from ptvrajsk/DG

UG & PPP Update

jetkan-yk committed:
Refactor to PDF++

WeiTangLau committed:
(v1.4) Activity Diagram for DeleteCommand

Created the activity diagram for DeleteCommand. Please check that it is
okay before putting the documentation for DeleteCommand.

jetkan-yk committed:
Merge branch 'master' of github.com:CS2103-AY1819S2-T12-4/main

WeiTangLau committed:
Merge branch 'master' into v1.4PPP

jetkan-yk committed:
Replace all persons to pdfs

WeiTangLau committed:
Travis check

ptvrajsk committed:
Merge pull request #310 from WeiTangLau/v1.4PPP

(v1.4) Diagrams for DG

jetkan-yk committed:
Update command result messages

jetkan-yk committed:
Checkstyle fixes

jetkan-yk committed:
Checkstyle fixes

jetkan-yk committed:
Fix failed tests

lormee94 committed:
(v1.4) DG sequence diagram standardised
- Merge
- Find

lormee94 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main into DG

lormee94 committed:
Merge pull request #311 from jetkan-yk/master

Final refactor to PDF++

ptvrajsk committed:
(v1.4) UG Completed

lormee94 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main into DG

ptvrajsk committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main

ptvrajsk committed:
UG Update

lormee94 committed:
Merge pull request #313 from ptvrajsk/master

UG Completion

lormee94 committed:
(v1.4) Update OpenCommandActivity Diagram

jetkan-yk committed:
Update sort deadline function

jetkan-yk committed:
Add SortCommand Autotest

jetkan-yk committed:
Checkstyle fixes

jetkan-yk committed:
Checkstyle fixes

WeiTangLau committed:
Instructions for manual testing in DG

Provided instructions for manual testing in DG.

lormee94 committed:
(v1.4) Rename Activity Diagram + changes to RenameCommand

WeiTangLau committed:
Travis check

lormee94 committed:
travis

lormee94 committed:
Travis

lormee94 committed:
Update RenameCommand.java

lormee94 committed:
Merge pull request #315 from WeiTangLau/v1.4Instructions

(v1.4) Instructions for manual testing in DG

ptvrajsk committed:
Merge pull request #312 from lormee94/DG

(v1.4) DG update of sequence diagrams

ptvrajsk committed:
Deadline ClassDiag and UseCase

ptvrajsk committed:
Merge branch 'master' into DG

ptvrajsk committed:
Use Case + DeadlineClassDiag

lormee94 committed:
Merge pull request #316 from ptvrajsk/DG

DG Update

jetkan-yk committed:
Update Deadline Command in DG

jetkan-yk committed:
Merge DG

ptvrajsk committed:
Merge pull request #314 from jetkan-yk/master

Update sort deadline function

ptvrajsk committed:
PPP Update

ptvrajsk committed:
Merge branch 'master' into DG

jetkan-yk committed:
Update diagrams for DeadlineCommand in DG

jetkan-yk committed:
Checkstyle fixes

jetkan-yk committed:
Merge branch 'master' of github.com:CS2103-AY1819S2-T12-4/main

ptvrajsk committed:
Grammer Fixes

lormee94 committed:
Merge pull request #318 from jetkan-yk/master

Update diagrams for DeadlineCommand in DG

lormee94 committed:
Merge pull request #317 from ptvrajsk/DG

PPP Update

ptvrajsk committed:
PPP Update

ptvrajsk committed:
PPP Update

ptvrajsk committed:
Fix

ptvrajsk committed:
History Command UG

ptvrajsk committed:
History Fix UG

ptvrajsk committed:
DG Fixes

jetkan-yk committed:
Update PPP

ptvrajsk committed:
UG Update

WeiTangLau committed:
(v1.4) weitanglau PPP

Provided the PPP.

lormee94 committed:
Merge pull request #319 from ptvrajsk/DG

PPP Update

jetkan-yk committed:
Update Jetkan-yk PPP

jetkan-yk committed:
Merge branch 'master' of github.com:CS2103-AY1819S2-T12-4/main

WeiTangLau committed:
Merge branch 'master' into v1.4Final

jetkan-yk committed:
Update AboutUs.adoc

ptvrajsk committed:
Merge pull request #320 from WeiTangLau/v1.4Final

(v1.4) weitanglau PPP

ptvrajsk committed:
Merge pull request #321 from jetkan-yk/master

Update jetkan-yk PPP

WeiTangLau committed:
File check

WeiTangLau committed:
minor adjustments

lormee94 committed:
(v1.4) Jeremy PPP, DG, yada

lormee94 committed:
Merge branch 'master' of https://github.com/cs2103-ay1819s2-t12-4/main into DG

WeiTangLau committed:
minor adjustments

jetkan-yk committed:
Fix typos

jetkan-yk committed:
Fix conflict

lormee94 committed:
Merge pull request #322 from WeiTangLau/master

File check

ptvrajsk committed:
Merge pull request #323 from lormee94/DG

(v1.4) Jeremy PPP and more

jetkan-yk committed:
Delete jetkan-yk.pdf

jetkan-yk committed:
Fix conflict

jetkan-yk committed:
DG Link fixes

jetkan-yk committed:
(v1.4) Final

ptvrajsk committed:
Merge pull request #324 from jetkan-yk/master

Final grammar fixes

jetkan-yk committed:
Final

jetkan-yk committed:
Final final

ptvrajsk committed:
Merge pull request #326 from jetkan-yk/master

Final

jetkan-yk committed:
Update UserGuide.adoc
